lib/net/http/header.rb

```diff
@@ -101,41 +258,63 @@ def add_field(key, val)
     end
   end
 
-  # [Ruby 1.8.3]
-  # Returns an array of header field strings corresponding to the
-  # case-insensitive +key+.  This method allows you to get duplicated
-  # header fields without any processing.  See also #[].
+  # Returns the array field value for the given +key+,
+  # or +nil+ if there is no such field;
+  # see {Fields}[rdoc-ref:Net::HTTPHeader@Fields]:
   #
-  #   p response.get_fields('Set-Cookie')
-  #     #=> ["session=al98axx; expires=Fri, 31-Dec-1999 23:58:23",
-  #          "query=rubyscript; expires=Fri, 31-Dec-1999 23:58:23"]
-  #   p response['Set-Cookie']
-  #     #=> "session=al98axx; expires=Fri, 31-Dec-1999 23:58:23, query=rubyscript; expires=Fri, 31-Dec-1999 23:58:23"
+  #   req = Net::HTTP::Get.new(uri)
+  #   req[:foo] = 'bar'
+  #   req.get_fields(:foo)    # => ["bar"]
+  #   req.add_field(:foo, 'baz')
+  #   req.get_fields(:foo)    # => ["bar", "baz"]
+  #   req.get_fields(:nosuch) # => nil
   #
   def get_fields(key)
     stringified_downcased_key = key.downcase.to_s
     return nil unless @header[stringified_downcased_key]
     @header[stringified_downcased_key].dup
   end
 
-  # Returns the header field corresponding to the case-insensitive key.
-  # Returns the default value +args+, or the result of the block, or
-  # raises an IndexError if there's no header field named +key+
-  # See Hash#fetch
+  # :call-seq
+  #   fetch(key, default_val = nil) {|key| ... }
+  #
+  # With a block, returns the string value for +key+ if it exists;
+  # otherwise returns the value of the block;
+  # see {Fields}[rdoc-ref:Net::HTTPHeader@Fields]:
+  #
+  #   req = Net::HTTP::Get.new(uri)
+  #   req[:foo] = 'bar'
+  #   req.fetch(:foo) {|key| key.capitalize }    # => "bar"
+  #   req.fetch(:nosuch) {|key| key.capitalize } # => "Nosuch"
+  #
+  # With no block, returns the string value for +key+ if it exists;
+  # otherwise, returns +default_val+ if it was given;
+  # otherwise raises an exception:
+  #
+  #   req.fetch(:foo)          # => "bar"
+  #   req.fetch(:nosuch, :baz) # => :baz
+  #   req.fetch(:nosuch)       # Raises IndexError.
+  #
   def fetch(key, *args, &block)   #:yield: +key+
     a = @header.fetch(key.downcase.to_s, *args, &block)
     a.kind_of?(Array) ? a.join(', ') : a
   end
 
-  # Iterates through the header names and values, passing in the name
-  # and value to the code block supplied.
+  # Calls the block with each key/value pair;
+  # returns the value of #to_hash;
+  # see {Fields}[rdoc-ref:Net::HTTPHeader@Fields]:
   #
-  # Returns an enumerator if no block is given.
+  #   req = Net::HTTP::Get.new(uri)
+  #   req.each_header {|key, value| p [key, value] }
   #
-  # Example:
+  # Output:
   #
-  #     response.header.each_header {|key,value| puts "#{key} = #{value}" }
+  #   ["accept-encoding", "gzip;q=1.0,deflate;q=0.6,identity;q=0.3"]
+  #   ["accept", "*/*"]
+  #   ["user-agent", "Ruby"]
+  #   ["host", "jsonplaceholder.typicode.com"]
   #
+  # Returns an enumerator if no block is given.
```

Comment:
Should we say that `each` is an alias of `each_header`?

Reply:
All aliases now noted.